File: app/src/main/java/com/rangerdie/runzombiez/mission/MissionRepository.kt
package com.rangerdie.runzombiez.mission

import android.content.Context
import kotlinx.serialization.json.Json

/**
 * Loads bundled mission JSON from assets/missions/. All MVP missions ship
 * locally with the APK — offline-first, no network required (spec section 13).
 */
class MissionRepository(private val context: Context) {

    private val json = Json { ignoreUnknownKeys = true }

    fun loadMission(assetFileName: String): Mission {
        val raw = context.assets.open("missions/$assetFileName").bufferedReader().use { it.readText() }
        return json.decodeFromString(Mission.serializer(), raw)
    }

    fun listMissionFiles(): List<String> =
        context.assets.list("missions")?.filter { it.endsWith(".json") }.orEmpty()
}

File: js/mission-repository.js
// Loads bundled mission JSON. Offline-first: missions/ is cached by the
// service worker, so this works with no network after first load.
export class MissionRepository {
  constructor(basePath = "missions/") {
    this.basePath = basePath;
  }

  async loadMission(fileName) {
    const response = await fetch(this.basePath + fileName);
    if (!response.ok) throw new Error(`Could not load mission "${fileName}": ${response.status}`);
    return response.json();
  }
}
